dataplicity/portforward.py

```diff
@@ -256,6 +256,7 @@ def __init__(self, client):
         self._client = weakref.ref(client)
         self._services = {}
         self._ports = {}
+        self._dynamic_ports = {}
```

Comment:
So this is here so that there is a reference to the service object?

But once added, nothing ever deletes the service from `_dynamic_ports`?

Reply:
yeah, this must be a reminescense of the times when the dynamic services were not launching a thread directly :D Great spot!